src/ray/gcs/tables.cc

```diff
@@ -438,6 +440,40 @@ std::string ClientTable::DebugString() const {
   return result.str();
 }
 
+Status ActorCheckpointIdTable::AddCheckpointId(const JobID &job_id,
+                                                  const ActorID &actor_id,
+                                                  const UniqueID &checkpoint_id) {
+  auto lookup_callback = [this, checkpoint_id, job_id, actor_id](
+                             ray::gcs::AsyncGcsClient *client, const UniqueID &id,
+                             const ActorCheckpointIdDataT &data) {
+    std::shared_ptr<ActorCheckpointIdDataT> copy =
+        std::make_shared<ActorCheckpointIdDataT>(data);
+    copy->timestamps.push_back(current_sys_time_ms());
+    copy->checkpoint_ids.push_back(checkpoint_id.binary());
+    auto num_to_keep = RayConfig::instance().num_actor_checkpoints_to_keep();
+    while (copy->timestamps.size() > num_to_keep) {
+      // Delete the checkpoint from actor checkpoint table.
+      const auto &checkpoint_id = UniqueID::from_binary(*copy->checkpoint_ids.begin());
+      RAY_LOG(DEBUG) << "Deleting checkpoint " << checkpoint_id << " for actor " << actor_id;
+      copy->timestamps.erase(copy->timestamps.begin());
+      copy->checkpoint_ids.erase(copy->checkpoint_ids.begin());
```

Comment:
Since checkpoint_ids is a long string concatenated objects:
```
checkpoint_ids: [string];
```
we should not erase the `begin()`.
`begin()` is just the first char of the string.

Reply:
Currently, `checkpoint_ids` is defined as a list of strings, using `begin()` is fine. But I should concatenate them as one single string. Thanks for reminding